DevelopmentProposals/reporterColumnLabels/reporterColumnLabels.md

```diff
@@ -0,0 +1,148 @@
+With the current design, it is impossible to determine whether the stored annotation was specified by the user or set by default.
+The user could even have specified a string that happens to be identical to the default.
+
+Currently, the ConsoleReporter always uses the annotation as the column label.
+This behavior appears to be appropriate for the ConsoleReporter because Output names tend to be short.
+Thus, if no annotation is specified, a line like
+```cpp
+reporter->updInput("inputs").connect(elbow->getCoordinate(PinJoint::Coord::RotationZ).getOutput("value"));
+```
+uses a default column label of "value"; if desired, the user can simply specify an annotation and re-run his/her code.
+There will likely be only a few Outputs connected to (the Input of) a ConsoleReporter because of the limited width of the typical console.
+(If too many Outputs are connected, the data corresponding to each time point will wrap over multiple lines and will be difficult to read.)
+Thus, providing an annotation if the default column label is unsatisfactory (e.g., "value" for all Coordinates) is not likely to be burdensome for users.
+
+The TableReporter always uses the (current) full path name as the column label.
+TableReporters are likely to report more Outputs than ConsoleReporters because, unlike the console,
+there is no detriment to writing a table with many columns (aside, of course, from memory issues for *very* large tables).
+As such, it is more important for the TableReporter to use meaningful default column labels.
+If reporting the value of all Coordinates in a Model, for example, it would be unhelpful for all column labels to be "value",
+and forcing the user to specify an annotation for each Output may not be practical.
+
+### Requirements
+- The ConsoleReporter and TableReporter should both use the annotation for the column label if an annotation has been provided by the user.
+- The ConsoleReporter should use a default column label that is likely to be as meaningful as possible while still being short.
+  Although not strictly enforced, the Output names tend to be short (and are automatically truncated to 12 characters if they are too long).
+- The TableReporter should use a default column name that is very likely to be meaningful.
+
+### Architecture
+Several designs would satisfy the requirements; four are proposed here.
+As a general note, it may be useful to add a `setAnnotation()` method (e.g., if a fully connected model was loaded from file).
+
+**Design 1**
+
+Store an empty string in `_annotations` if no annotation was provided when `Input::connect()` was called.
+When writing the column label, use the `_annotations` value if it is not empty; otherwise,
+the ConsoleReporter will use the Output's name and the TableReporter will use the Output's full path name.
+This design will change the current behavior.
+Currently, the name of the Output can differ from the column label if the name is changed after `Input::connect()` has been called.
```

Comment:
Changing names currently is a change in a Component's properties. That should invalidate all connections and require a connect() to be invoked.